components/job-orchestration/job_orchestration/retention/utils.py

```
@@ -0,0 +1,151 @@
+
+    def get_targets(self) -> Set[str]:
+        return self._targets
+
+    def persists_new_targets(self) -> None:
+        """
+        Writes any new targets added since initialization to the recovery file.
+        """
+        if len(self._targets_to_persist) == 0:
+            return
+
+        with open(self._recovery_file_path, "a") as f:
```

Pylint (3.3.7)

[warning] Using open without explicitly specifying an encoding

(W1514)